Alloy/Shaders/Shaders.metal

```diff
@@ -11,18 +11,167 @@
 
 using namespace metal;
 
-constant bool deviceSupportsFeaturesOfGPUFamily4_v1 [[function_constant(0)]];
+constant bool device_supports_features_of_gpu_family4_v1 [[function_constant(0)]];
 
-kernel void textureCopy(texture2d<half, access::read> texture1 [[ texture(0) ]],
-                        texture2d<half, access::write> texture2 [[ texture(1) ]],
+struct BlockSize {
+    ushort width;
+    ushort height;
+};
+
+kernel void textureCopy(texture2d<half, access::read> texture_1 [[ texture(0) ]],
+                        texture2d<half, access::write> texture_2 [[ texture(1) ]],
                         const ushort2 position [[thread_position_in_grid]]) {
-    if (!deviceSupportsFeaturesOfGPUFamily4_v1) {
-        if (position.x >= texture1.get_width() || position.y >= texture1.get_height()) {
+    if (!device_supports_features_of_gpu_family4_v1) {
+        if (position.x >= texture_1.get_width() || position.y >= texture_1.get_height()) {
             return;
         }
     }
 
-    const half4 c1 = texture1.read(position);
+    const half4 c_1 = texture_1.read(position);
+
+    texture_2.write(c_1, position);
+}
+
+kernel void max(texture2d<half, access::sample> input_texture [[ texture(0) ]],
+                constant BlockSize& block_size [[ buffer(0) ]],
+                device float4& result [[ buffer(1) ]],
+                threadgroup half4* shared_memory [[ threadgroup(0) ]],
+                const ushort tiit [[ thread_index_in_threadgroup ]],
+                const ushort2 tpig [[ thread_position_in_grid ]],
+                const ushort2 tpt [[ threads_per_threadgroup ]]) {
```

Comment:
let's keep longer names for shared meta data (i.e. not per-thread)

Reply:
I think in this case we should use long obvious names for everything to keep the code in a single style